### ISSUE TITLE
No cabal file found for Archive

### ISSUE THREAD

**Issue by waddlaw** [commercialhaskell/stack#4292]
### Steps to reproduce

1. create new project.

```sh
$ stack new sample
$ cd sample
```

2. edit `stack.yaml` (add `extra-deps`)

```sh
$ cat stack.yaml
resolver:
  url: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/12/9.yaml

packages:
- .
extra-deps:
- github: fpco/store
  commit: 247f2863bed0f5c1003ab835075b6dcafb9aa773
```

3. build project

```sh
$ stack build
No cabal file found for Archive from https://github.com/fpco/store/archive/247f2863bed0f5c1003ab835075b6dcafb9aa773.tar.gz
```

Some packages will succeed.(ex. `cassava`)

```yaml
extra-deps:
- git: https://github.com/haskell-hvr/cassava.git
  commit: 151f74b976ce93c59e938ae130324bec53ac8c92
# - github: fpco/store
#   commit: 247f2863bed0f5c1003ab835075b6dcafb9aa773
```

### Stack version

```
$ stack --version
Version 1.10.0, Git revision 388232c3f3ae350e384b2e41c0c11606fd06b69e (6449 commits) x86_64 hpack-0.30.0
```

**Comment by brandon-leapyear** [comment]
Also breaks if you include
```
subdirs:
- .
```

Error is in Pantry.Archive stripping prefix of a path instead of stripping subdirectories. e.g. specifying
```
subdirs:
- foo/bar
```
in a repository with the path `foo/bar-asdf.cabal`, it will resolve this as `-asdf.cabal`

**Comment by waddlaw** [comment]
@brandon-leapyear @mihaimaruseac 
thanks! But, still unsolved problem...

I think that `parseArchive` is cause.

https://github.com/commercialhaskell/stack/blob/master/subs/pantry/src/Pantry/Archive.hs#L392

```hs
...
(cabalPath, cabalEntry@(TreeEntry cabalBlobKey _)) <- findCabalFile pli tree
..
```

With this implementation, it's necessary `.cabal` file in the archive. (don't auto generate from `package.yaml`)

We must to contains `.cabal` files in archive, even if exist `package.yaml`?

**Comment by mihaimaruseac** [comment]
Reopening in that case, the file should be regenerated

**Comment by snoyberg** [comment]
Please see #4377.

**Comment by snoyberg** [comment]
Fixed by #4439
